packages/@aws-cdk/aws-redshift/lib/cluster-ref.ts

```diff
@@ -0,0 +1,52 @@
+import * as ec2 from '@aws-cdk/aws-ec2';
+import * as secretsmanager from '@aws-cdk/aws-secretsmanager';
+import { IResource } from '@aws-cdk/core';
+import { Endpoint } from './endpoint';
+
+/**
+ * Create a Redshift Cluster with a given number of nodes.
+ */
+export interface ICluster extends IResource, ec2.IConnectable, secretsmanager.ISecretAttachmentTarget {
+  /**
+   * Name of the cluster
+   * @attribute ClusterName
+   */
+  readonly clusterName: string;
+
+  /**
+   * The endpoint to use for read/write operations
+   * @attribute EndpointAddress,EndpointPort
+   */
+  readonly clusterEndpoint: Endpoint;
+
+  /**
+   * The security group for this database cluster
+   */
+  readonly securityGroupIds: string[];
+}
+
+/**
+ * Properties that describe an existing cluster instance
+ */
+export interface ClusterAttributes {
+  /**
+   * The security groups of the redshift cluster
+   */
+  readonly securityGroups: ec2.ISecurityGroup[];
```

Comment:
Is this really required? Feels like it should be optional (empty by default), but maybe I'm missing something.

Reply:
Yeah I guess that makes sense. However, I still have some blind spots about what importing resources via `fromResourceAttributes` should accomplish. Like what should I be able to to with the imported resource? Should I be able to import "incomplete" resources?

I was thinking, since `ICluster` implements `ec2.IConnectable` it would not make much sense importing it without the security groups. But then again you could have a cluster without attached security groups. :thinking: 

Is there maybe some piece of documentation or a discussion where the concepts behind importing resources are made somewhat clearer, so I can read up on that?

Reply:
Looking at similar resources:

1. [`IProject` implement `IConnectable`](https://github.com/aws/aws-cdk/blob/4fe0e3b7a638eae2135facdc4283cbd136c7b44c/packages/%40aws-cdk/aws-codebuild/lib/project.ts#L24-L24), but [doesn't allow specifying the security groups](https://github.com/aws/aws-cdk/blob/4fe0e3b7a638eae2135facdc4283cbd136c7b44c/packages/%40aws-cdk/aws-codebuild/lib/project.ts#L565-L596).

2. [`IFunction` implements `IConnectable`](https://github.com/aws/aws-cdk/blob/4fe0e3b7a638eae2135facdc4283cbd136c7b44c/packages/%40aws-cdk/aws-lambda/lib/function-base.ts#L12-L12), but [has optional security group](https://github.com/aws/aws-cdk/blob/4fe0e3b7a638eae2135facdc4283cbd136c7b44c/packages/%40aws-cdk/aws-lambda/lib/function-base.ts#L142-L142).

So definitely make it optional at the very least (I would also be OK with removing his for now).